### DISCUSSION TITLE
Datestring format mismatch between Send at and the date picker

### DISCUSSION THREAD

**Discussion by lkarabeg** [Extended-Thunder/send-later#433] [General]
Steps to reproduce:

1. Open the Send Later dialog
2. Press Tab to go from Send At to the date picker (the Send at field will update)
3. Press Shift + Tab to go back to Send at (the datepicker will now update with a new date)

![image](https://user-images.githubusercontent.com/23399564/145458064-0ac4b861-910b-490a-b7ba-f7b1c136b493.png)
![image](https://user-images.githubusercontent.com/23399564/145458110-99f4fa3f-1eeb-4db1-8c36-ff3829948a8c.png)
![image](https://user-images.githubusercontent.com/23399564/145460203-7afec2ec-caed-47db-9411-34b0ec64a08b.png)

What appears to be happening is that the date picker is providing a little-endian date, like dd.MM.yyyy which is being misinterpreted as a middle-endian date, like M.dd.yyyy. 

Another way to demonstrate the problem is to select a date and click the "Save these values as the defaults" button. After closing and reopening the dialog, the wrong date will be set as default.

I'm running Windows, and my system date format is yyyy-MM-dd. My more general format preference is set to "Norwegian bokmål", which I would guess might be determining the date picker format. 

I've used Send later for a very long time, and the problem just appearred recently, so I suspect it's a regression in version 9.06.

**Comment by lkarabeg** [comment]
I may have accidentally duplicated #401, but I'll leave this here as it's a bit better explained IMO.

**Comment by lkarabeg** [comment]
The problem goes away when I set a custom short date format in the options. Now saving defaults works, and the date picker and Send at field agree after the dates have been reinterpreted as shown above.

**Comment by jph-sendlater** [comment]
Date-time parsing/formatting has always been an issue. It seems like fixing one inconsistent behavior always somehow breaks another one :roll_eyes:. Anyway, I'm glad the custom date/time format feature seems to be working for you. Let me know if you run into any more issues.

**Comment by jikamens** [comment]
It took me pretty much all day today, but I believe I both figured out the root cause of the text box and the date picker being out of sync with each other with regards to the proper date format, _and_ figured out a fix that should work for most people. I've implemented that fix in [release 10.0.5](https://github.com/Extended-Thunder/send-later/releases/tag/v10.0.5). If you'd like you can download the pre-release from there and confirm that it fixes the problem for you. In fact, I'd **really** appreciate it if you did that, because I'd like confirmation that the fix actually works before I post the release to addons.thunderbird.net.

You can download the XPI file from the release page and then open the add-ons page in Thunderbird, click on the gear icon, select 'Install add-on from file", and browse to the downloaded XPI file.

I've also documented this problem in detail and how to work around it using a custom date format for the rare cases in which the fix I just implemented doesn't do the trick. See the [user guide](https://extended-thunder.github.io/send-later/#date-format-problems).

**Comment by jikamens** [comment]
Closing because this is fixed.
